avahi-core/announce.c

```diff
@@ -44,6 +44,12 @@ static void remove_announcer(AvahiServer *s, AvahiAnnouncer *a) {
     AVAHI_LLIST_REMOVE(AvahiAnnouncer, by_interface, a->interface->announcers, a);
     AVAHI_LLIST_REMOVE(AvahiAnnouncer, by_entry, a->entry->announcers, a);
 
+    if (a->state == AVAHI_PROBING && a->entry->group) {
+	assert(a->entry->group->n_probing);
+	a->entry->group->n_probing--;
+	avahi_s_entry_group_check_probed(a->entry->group, 1);
```

Comment:
I'd propose deferring any announcements by doing `avahi_s_entry_group_check_probed(a->entry->group, 0);` since some or all of the remaining announcers in the entry group may be about to be removed as well, and their goodbyes have already been sent